src/lib/ng-package/entry-point/write-package.transform.ts

```diff
@@ -185,7 +185,7 @@ async function writePackageJson(
   try {
     checkNonPeerDependencies(packageJson, 'dependencies', whitelist, spinner);
   } catch (e) {
-    await rimraf(entryPoint.destinationPath);
+    await rimraf(entryPoint.libraryDestinationPath);
```

Comment:
@alan-agius4, it turns out the output directory contain the most of the files for the secondary entry point, but not the directory with its package.json. This PR fixes the issue and adds tests.

Once this is merged, I'll continue with the #1737.